tests/integration/test_simulation.py

```diff
@@ -30,96 +30,260 @@ def get_msg_queue_item(queue, id_):
+            id='do-not-fail-this-cp'),
+        # Task fails this CP, first submit.
+        param(
+            'fail_once', '1068', (True, False, False),
+            id='and-another-cp'),
+        # Task fails this CP, all submits.
+        param(
+            'fail_all_submits', '1066', (True, True, True),
+            id='fail-all-submits'),
+        # Task succeeds this CP, all submits.
+        param(
+            'fail_all_submits', '1067', (False, False, False),
+            id='fail-no-submits'),
+    )
+)
+def test_fail_once(sim_time_check_setup, itask, point, results):
+    """A task with a fail cycle point only fails
+    at that cycle point, and then only on the first submission.
+    """
+    schd, _, msg_q = sim_time_check_setup
+
+    itask = schd.pool.get_task(
+        ISO8601Point(point), itask)
+
+    for result in results:
+        schd.task_job_mgr._simulation_submit_task_jobs(
+            [itask], schd.workflow)
+        assert itask.mode_settings.sim_task_fails is result
+
+
+async def test_sim_time_check_sets_started_time(
+    scheduler, sim_time_check_setup
+):
+    """But sim_time_check still returns False
+
+    This only occurs in reality if we've restarted from database and
+    not retrieved the started time from itask.summary.
+    """
     schd, _, msg_q = sim_time_check_setup
     one_1066 = schd.pool.get_task(ISO8601Point('1066'), 'one')
-    one_1066.state.status = 'running'
+    schd.task_job_mgr._simulation_submit_task_jobs(
+        [one_1066], schd.workflow)
+    schd.workflow_db_mgr.process_queued_ops()
+    one_1066.summary['started_time'] = None
+    one_1066.state.is_queued = False
     assert one_1066.summary['started_time'] is None
-    assert sim_time_check(msg_q, [one_1066]) is False
+    assert sim_time_check(
+        msg_q, [one_1066], schd.task_events_mgr.broadcast_mgr,
+        schd.workflow_db_mgr
+    ) is False
     assert one_1066.summary['started_time'] is not None
 
 
 def test_task_finishes(sim_time_check_setup, monkeypatch):
     """...and an appropriate message sent.
 
-    Checks all possible outcomes in sim_time_check where elapsed time is
-    greater than the simulation time.
+    Checks that failed and bar are output if a task is set to fail.
 
-    Does NOT check every possible cause on an outcome - this is done
+    Does NOT check every possible cause of an outcome - this is done
     in unit tests.
     """
     schd, _, msg_q = sim_time_check_setup
     monkeypatch.setattr('cylc.flow.simulation.time', lambda: 0)
 
-    # Setup a task to fail
+    # Setup a task to fail, submit it.
     fail_all_1066 = schd.pool.get_task(ISO8601Point('1066'), 'fail_all')
     fail_all_1066.state.status = 'running'
-    fail_all_1066.try_timers = {'execution-retry': SimpleNamespace(num=0)}
+    fail_all_1066.state.is_queued = False
+    schd.task_job_mgr._simulation_submit_task_jobs(
+        [fail_all_1066], schd.workflow)
+
+    # For the purpose of the test delete the started time set but
```

Comment:
Is this meant to be?
```suggestion
    # For the purpose of the test delete the started time set by
```